xkafka/consumer.go

```diff
@@ -77,57 +84,135 @@ func (c *Consumer) Start(ctx context.Context) error {
 
 	c.handler = c.concatMiddlewares(c.handler)
 
-	st := stream.New().WithMaxGoroutines(c.config.concurrency)
-	errChan := make(chan error, 1)
+	if c.config.concurrency > 1 {
+		return c.runAsync(ctx)
+	}
+
+	return c.runSequential(ctx)
+}
+
+func (c *Consumer) runSequential(ctx context.Context) (err error) {
+	defer func() {
+		if uerr := c.unsubscribe(); uerr != nil {
+			err = errors.Join(err, uerr)
+		}
+	}()
 
 	for {
 		select {
 		case <-ctx.Done():
-			st.Wait()
+			return err
+		default:
+			km, err := c.kafka.ReadMessage(c.config.pollTimeout)
+			if err != nil {
+				var kerr kafka.Error
+				if errors.As(err, &kerr) && kerr.Code() == kafka.ErrTimedOut {
+					continue
+				}
+
+				if ferr := c.config.errorHandler(err); ferr != nil {
```

Comment:
When `errorHandler` return `nil` from here. The iteration is not skipped, and it invokes `newMessage` causing nil deref, ideally in-case handler returns nil at this point, iteration should be skipped (with `continue`).

Reply:
Example:
```go
xkafka.ErrorHandler(func(err error) error {
	var kerr kafka.Error
	if errors.As(err, &kerr) && kerr.Code() == kafka.ErrUnknownTopicOrPart {
		return nil
	}

	return err
})
```

Or 
```go
xkafka.ErrorHandler(func(err error) error {
	var kerr kafka.Error
	if errors.As(err, &kerr) && !kerr.IsFatal() {
		return nil
	}

	return err
})
```